File: inetpub/wwwroot/SixtServiceLeas/SixtServiceLeas/Lib/RequestLoggingModule.vb
Imports System.IO

Public Class RequestLoggingModule
    Implements IHttpModule

    Public Sub Dispose() Implements IHttpModule.Dispose

    End Sub

    Public Sub Init(context As HttpApplication) Implements IHttpModule.Init
        AddHandler context.BeginRequest, AddressOf App_BeginRequest
        AddHandler context.EndRequest, AddressOf App_EndRequest
    End Sub

    Public Sub App_BeginRequest(ByVal source As Object, ByVal e As EventArgs)
        Dim app As HttpApplication = CType(source, HttpApplication)
        Dim context As HttpContext = app.Context

        If context.Request.ContentLength = 0 Then Return

        Dim request As HttpRequest = context.Request
        Dim response As HttpResponse = context.Response

        Dim filter As New ResponseFilterStream(response.Filter)
        response.Filter = filter

        Dim logService As GeneralTools.Services.LogService = New GeneralTools.Services.LogService(String.Empty, String.Empty)
        logService.LogWebServiceTraffic("Request", GetString(request.InputStream), ConfigurationManager.AppSettings("LogTableName"))
    End Sub

    Public Sub App_EndRequest(ByVal source As Object, ByVal e As EventArgs)
        Dim app As HttpApplication = CType(source, HttpApplication)
        Dim context As HttpContext = app.Context

        If context.Request.ContentLength = 0 Then Return

        Dim filter As ResponseFilterStream = CType(context.Response.Filter, ResponseFilterStream)

        Dim logService As GeneralTools.Services.LogService = New GeneralTools.Services.LogService(String.Empty, String.Empty)
        logService.LogWebServiceTraffic("Response", filter.ReadStream(), ConfigurationManager.AppSettings("LogTableName"))
    End Sub

    Private Function GetString(ByVal stream As Stream) As String
        Dim bytes = New Byte(stream.Length - 1) {}
        stream.Read(bytes, 0, bytes.Length)
        stream.Position = 0
        Return Encoding.UTF8.GetString(bytes)
    End Function

End Class

File: Work/WebServices/SoapRuecklaeuferschnittstelle/SoapRuecklaeuferschnittstelle/RequestLoggingModule.cs
using System;
using System.Configuration;
using System.IO;
using System.Text;
using System.Web;
using GeneralTools.Services;

namespace SoapRuecklaeuferschnittstelle
{
    public class RequestLoggingModule : IHttpModule
    {
        public void Dispose()
        {
            
        }

        public void Init(HttpApplication context)
        {
            context.BeginRequest += context_BeginRequest;
            context.EndRequest += context_EndRequest;
        }

        void context_BeginRequest(object sender, EventArgs e)
        {
            var app = sender as HttpApplication;
            var context = app.Context;

            if (context.Request.ContentLength == 0)
                return;

            var request = context.Request;
            var response = context.Response;

            var filter = new ResponseFilterStream(response.Filter);
            response.Filter = filter;

            var logService = new LogService(String.Empty, String.Empty);
            logService.LogWebServiceTraffic("Request", GetString(request.InputStream), ConfigurationManager.AppSettings["LogTableName"]);
        }

        void context_EndRequest(object sender, EventArgs e)
        {
            var app = sender as HttpApplication;
            var context = app.Context;

            if (context.Request.ContentLength == 0)
                return;

            var filter = context.Response.Filter as ResponseFilterStream;

            var logService = new LogService(String.Empty, String.Empty);
            logService.LogWebServiceTraffic("Response", filter.ReadStream(), ConfigurationManager.AppSettings["LogTableName"]);
        }

        private string GetString(Stream stream)
        {
            var bytes = new Byte[stream.Length - 1];
            stream.Read(bytes, 0, bytes.Length);
            stream.Position = 0;
            return Encoding.UTF8.GetString(bytes);
        }
    }
}